sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppImpl.java

```diff
@@ -415,7 +416,17 @@ public FunctionAppImpl withPrivateDockerHubImage(String imageAndTag) {
     @Override
     public FunctionAppImpl withPrivateRegistryImage(String imageAndTag, String serverUrl) {
         ensureLinuxPlan();
-        return super.withPrivateRegistryImage(imageAndTag, serverUrl);
+        super.withPrivateRegistryImage(imageAndTag, serverUrl);
+        if (isFunctionAppOnACA()) {
+            try {
+                URL url = new URL(serverUrl);
+                // remove URL protocol, as ACA don't allow that
+                withAppSetting(SETTING_REGISTRY_SERVER, url.getAuthority() + url.getFile());
```

Comment:
What if the serverUrl already removed the protocol? Will `new URL` report error?

Reply:
Yeah, it will. When this happens, we will silently ignore the error and sets the `SETTING_REGISTRY_SERVER` to the given serverUrl in `super.withPrivateRegistryImage()`:
https://github.com/Azure/azure-sdk-for-java/blob/17e41975b806fb06b92656570fb04f1e17ac9d2b/sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/AppServiceBaseImpl.java#L478